libtenzir/builtins/formats/lines.cpp

```diff
@@ -0,0 +1,148 @@
+#include <tenzir/plugin.hpp>
+#include <tenzir/table_slice_builder.hpp>
+#include <tenzir/to_lines.hpp>
+
+#include <optional>
+
+namespace tenzir::plugins::lines {
+
+namespace {
+
+/// The default name of the record field.
+const auto default_field_name = std::string{"data"};
+
+struct parser_args {
+  std::optional<located<std::string>> field_name;
+  std::optional<location> skip_empty;
+
+  template <class Inspector>
+  friend auto inspect(Inspector& f, parser_args& x) -> bool {
+    return f.object(x)
+      .pretty_name("parser_args")
+      .fields(f.field("field_name", x.field_name),
+              f.field("skip_empty", x.skip_empty));
+  }
+};
+
+auto line_type(std::string field_name = "data") -> type {
+  TENZIR_ASSERT(not field_name.empty());
+  return type{
+    "tenzir.line",
+    record_type{
+      {std::move(field_name), string_type{}},
+    },
+  };
+}
+
+class lines_parser final : public plugin_parser {
+public:
+  lines_parser() = default;
+
+  explicit lines_parser(parser_args args) : args_{std::move(args)} {
+  }
+
+  auto name() const -> std::string override {
+    return "lines";
+  }
+
+  auto
+  instantiate(generator<chunk_ptr> input, operator_control_plane& ctrl) const
+    -> std::optional<generator<table_slice>> override {
+    auto field_name
+      = args_.field_name ? args_.field_name->inner : default_field_name;
+    auto make
+      = [](auto& ctrl, generator<chunk_ptr> input, std::string field_name,
+           bool skip_empty) -> generator<table_slice> {
+      auto make_builder = [field_name = std::move(field_name)]() {
+        return table_slice_builder{line_type(field_name)};
+      };
+      auto num_non_empty_lines = size_t{0};
+      auto num_empty_lines = size_t{0};
+      auto builder = make_builder();
+      for (auto line : to_lines(std::move(input))) {
+        if (not line) {
+          co_yield {};
+          continue;
+        }
+        if (line->empty()) {
+          ++num_empty_lines;
+          if (skip_empty) {
+            co_yield {};
+            continue;
+          }
+        } else {
+          ++num_non_empty_lines;
+        }
+        auto num_lines = skip_empty ? num_non_empty_lines
+                                    : num_non_empty_lines + num_empty_lines;
+        if (not builder.add(*line)) {
+          diagnostic::error("failed add line")
+            .hint("line number: ", num_lines + 1)
+            .hint("{}", *line)
+            .emit(ctrl.diagnostics());
+          co_return;
+        }
+        // FIXME: make configurable. Like in PCAP?
+        //
+        //    if (builder.rows() >= defaults::import::table_slice_size
+        //        or last_finish + defaults::import::batch_timeout < now)
+        //
+        // To be discussed in review.
+        if (builder.rows() == 10) {
+          co_yield builder.finish();
+          builder = make_builder();
+        }
+      }
+      if (builder.rows() > 0)
+        co_yield builder.finish();
```

Comment:
```suggestion
        co_yield builder.finish("tenzir.lines");
```

For consistency with other operators and formats that have a schema without a name.

Reply:
Ah nvm, you're not using the adaptive builder and instead have a fixed schema above.